packages/drivers/odsp-driver/src/odspDeltaStorageService.ts

```diff
@@ -22,6 +23,12 @@ import { getWithRetryForTokenRefresh } from "./odspUtils";
  * Provides access to the underlying delta storage on the server for sharepoint driver.
  */
 export class OdspDeltaStorageService {
+    // This set is used to record the id for the request. The id is formed using the from/to of the ops fetch request.
+    // We don't want to put SPResponse guid in all ops fetch requests as it overrides the SPRequestGuid on server and
+    // then the server cannot determine the farm/time on which the request is executed. So for now we only want to
+    // use SpResponseGuid for requests that we think are retried requests. So if a request is made for same ops range
+    // then we put the SPResponseGuid in the fetch request.
+    private readonly deltaRequestIdSet = new Set();
```

Comment:
I do not think we care that much about the case you describe, at least not directly. We care about timeouts or infinite hangs or failed fetches where we do not get back SPO correlation ID, and thus can't figure out what actually happened (did it even get to SPO?). So I think the solution is to always provide client generated ID in addition to having existing SPO correlation ID logic being exactly the same (not changed by new logic). I.e. be it pure addition, not replacement.

We have had a brief discussion on that in SPO-Fluid Thu 10am forum and based on my recollection, that's exactly how Marcelo things about it too.

Reply:
Also note that deleting this.deltaRequestIdSet.add(id); at the bottom makes it supply ID on every successful call, so comment is not entirely correct - it implements retries where previous call failed. It likely to give us what we want but only if there is a second call for same thing, whereas I think we should get it every time, and for all calls (including TreesLatest), as not all calls are retried or have a chance to be retried.